packages/fork-choice/src/forkChoice/forkChoice.ts

```diff
@@ -156,6 +168,41 @@ export class ForkChoice implements IForkChoice {
     return this.head;
   }
 
+  /**
+   *
+   * A multiplexer to wrap around the traditional `updateHead()` according to the scenario
```

Comment:
as commented in [prepareNextSlot.ts](https://github.com/ChainSafe/lodestar/pull/6652/files#r1614398016), we may want `predictProposerHead()` to also return the weak head root + slot
then each mode below returns different results:
- predicted head + weak head if `GetPredictedProposerHead`
- head + isHeadTimely + notReorgedReason if `GetProposerHead`
- only head if `GetCanonicialHead`

consider separating to 3 apis if needed